applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -48,34 +55,43 @@ export class ReactNativePlugin extends BaseTelemetryPlugin {
                 core?: IAppInsightsCore,
                 extensions?: IPlugin[]
             ) => {
+                let identifier = this.identifier;
                 if (!_self.isInitialized()) {
                     _base.initialize(config, core, extensions);
 
-                    const inConfig = config || {};
-                    const defaultConfig = _getDefaultConfig();
-                    objForEachKey(defaultConfig, (option, value) => {
-                        _config[option] = ConfigurationManager.getConfig(
-                            inConfig as any,
-                            option,
-                            _self.identifier,
-                            !isUndefined(_config[option]) ? _config[option] : value
-                        );
-                    });
-        
-                    if (!_config.disableDeviceCollection) {
-                        _self._collectDeviceInfo();
-                    }
-        
-                    if (core && core.getPlugin) {
-                        _analyticsPlugin = core.getPlugin<any>(AnalyticsPluginIdentifier)?.plugin as IAppInsights;
-                    }
-        
-                    if (!_config.disableExceptionCollection) {
-                        _self._setExceptionHandler();
-                    }
+                    _self._addHook(onConfigChange(config, (details) => {
+                        let ctx = _self._getTelCtx();
+                        _config = ctx.getExtCfg<IReactNativePluginConfig>(identifier, defaultReactNativePluginConfig);
+
+                        if (!_config.disableDeviceCollection) {
+                            if (notInit){
+                                _self._collectDeviceInfo();
+                                notInit = false;
```

Comment:
A better name should be used as I first read `notInit` (above) as this component is not initialized)

Perhaps: `deviceIngoCollected` or something like that.